components/src/main/java/dev/arkbuilders/components/filepicker/ArkFilePickerFragment.kt

```diff
@@ -353,11 +376,10 @@ open class ArkFilePickerFragment :
         }
     } else {
         listOf(
-            FilesPage(this, viewModel, itemsPluralId!!)
+            FilesPage(this, viewModel, itemsPluralId!!, pinFileCallback = mPinFileCallback)
         )
     }
```

Comment:
I think we need to get rid of callback and move `pinFile` to viewModel since this is business logic.
Also, you don't have to create a class for each callback. Thanks to Kotlin we can use lambdas:
`fun someFunction(callback: () -> Unit) {}`